ssh/manifests/server/config_users.pp

```diff
@@ -0,0 +1,53 @@
+class ssh::server::config_users (
+  $users = $quickstack::params::moc_users
+) {
+
+  define config_user ($user_config = $title) {
```

Comment:
Input is specified in hiera file as:
quickstack::params::moc_users [
{
    username: ...,
    password: ...,
    hashed_password: ...,
}, 
{
    username: ...,
    password: ...,
    hashed_password: ...,
},
]

So, we pass an array of users, each user consists of a dictionary(python term, not sure what we call in puppet) and each dictionary contains username, password and hashed_password.

calling "config_user {users: }" iterates over the array calls the mentioned function on individual user. We extract the individual details inside the define method.